controllers/util/solr_util.go

```diff
@@ -1010,21 +992,21 @@ func generateZKInteractionInitContainer(solrCloud *solr.SolrCloud, solrCloudStat
 
 	cmd := ""
 
+	if hasChroot {
+		cmd += "solr zk ls ${ZK_CHROOT} -z ${ZK_SERVER} || solr zk mkroot ${ZK_CHROOT} -z ${ZK_SERVER}; "
+	}
+
 	if solrCloud.Spec.SolrTLS != nil {
 		cmd = setUrlSchemeClusterPropCmd()
```

Comment:
I think you want this to be a `+=`, if you are moving the `hasChroot` logic above here.